### PR TITLE
Resolved issue of Doublication of messages #2851

### LINKED ISSUE #2851: Doublication of messages
**Actual Behaviour**
When you ask "who is Mario" the answer is
2 times "I’d rather not say."
or
2 Times Why do you ask?

**Expected Behaviour**
Only once.

### PR DESCRIPTION
**Pull request by Vineet-Sharma29**
Fixes #2851 

Changes: Following changes have been made:-
1. in src/utils/generateMessage.js in line 39 forEach loop has been removed because in some cases multiple actions were obtained due to which susiReply were multiple
2. it is replaced by taking only the first entry from the actions array

Demo Link: https://pr-2911-fossasia-susi-web-chat.surge.sh

Screenshots of the change: 

**Before**

Screenshot from 2019-09-17 20-00-58

**After**

Screenshot from 2019-09-17 20-04-03

### CODE REVIEW

File: src/utils/generateMessage.js [hunk 1]
```diff
@@ -36,17 +36,16 @@ export default function({
...
-                  tasks.push(
-                    plannedFunction,
-                    () => createSusiMessage({ message: eachMessage }),
-                    eachMessage.planDelay,
-                  );
-                } else {
-                  createSusiMessage({ message: eachMessage });
-                }
-              });
+              if (susiMessages[0].planDelay) {
```
Comment (akshatnitd) [inline src/utils/generateMessage.js]: This is the incorrect fix, we need all the messages that comes in the `susiMessages` array. This piece of code was added to make Planned actions work.

Comment (Vineet-Sharma29) [reply src/utils/generateMessage.js]: ok @akshatnitd I will fix it removing duplicate entries from the `susiMessages` array and then runnnig `forEach` loop over the array so as to print all the messages. Can you suggest if this approach is correct?

Comment (akshatnitd) [reply src/utils/generateMessage.js]: Does `susiMessages` contain duplicate entries? Can you give an example case?

Comment (Vineet-Sharma29) [reply src/utils/generateMessage.js]: @akshatnitd when I have query : `who is mario` , I get following list of actions(as observed in networking tab of console):-
`
"actions": [
      {
        "language": "en",
        "type": "answer",
        "expression": "Great question. I never actually thought about that before, and I am not sure."
      },
      {
        "type": "answer",
        "mood": "sabta",
        "expression": "Great question. I never actually thought about that before, and I am not sure."
      }
    ],
`

Comment (Vineet-Sharma29) [reply src/utils/generateMessage.js]: @akshatnitd changes has been done so as to include non-duplicate, multiple messages

Comment (akshatnitd) [reply src/utils/generateMessage.js]: @Vineet-Sharma20 For this specifically, this should be fixed from backend and only 1 action should be ideally coming from backend.

Comment (Vineet-Sharma29) [reply src/utils/generateMessage.js]: @akshatnitd ok

Comment (Vineet-Sharma29) [reply src/utils/generateMessage.js]: @akshatnitd I have opened this as an issue in susi_server

Comment (akshatnitd) [reply src/utils/generateMessage.js]: @Vineet-Sharma29 Thanks! I think you can close this PR as it is a server issue!

**Pull request closed without merging**